common/tools/dev-tool/src/commands/run/typecheck.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as path from "node:path";
+import { leafCommand, makeCommandInfo } from "../../framework/command";
+import { createPrinter } from "../../util/printer";
+import { resolveProject } from "../../util/resolveProject";
+import { Project } from "ts-morph";
+import { DiagnosticCategory } from "typescript";
+
+const log = createPrinter("typecheck");
+
+export const commandInfo = makeCommandInfo(
+  "typecheck",
+  "typecheck typescript code files that are not part of tshy build",
+  {
+    paths: {
+      kind: "string",
+      description:
+        "relative path pattern of additional files (not included in tsconfig.json) to check for compilation errors. Examples: '--paths samples-dev/**/*.ts'.",
+    },
+  },
+);
+
+export default leafCommand(commandInfo, async (options) => {
+  log.info(`type-checking...`);
+  const { path: projPath } = await resolveProject(process.cwd());
+
+  const project = new Project({
+    tsConfigFilePath: path.join(projPath, "tsconfig.json"),
+  });
+
+  if (options.paths) {
```

Comment:
Could we allow for multiple paths such as test and samples-dev and some other random folder?

Reply:
yes, that makes sense. We could use a comma separated list.  test and samples-dev can be added to tsconfig.json too and will be used by this command.

@mpodwysocki do you remember why tshy doesn't report any errors from non-src folders even though the "includes" in tsconfig.json contains test/samples-dev?

Reply:
previously i read this [doc](https://github.com/isaacs/tshy/tree/main?tab=readme-ov-file#tsconfigs)
![image](https://github.com/user-attachments/assets/dd8f2773-afac-4a75-977f-2a00805a96a7)

Reply:
Updated to allow comma separated patterns